### ISSUE TITLE
tf.MonitoredSession accessing raw session raw_session like in SingularMonitoredSession

### ISSUE THREAD

**Issue by ricvo** [tensorflow/tensorflow#23403]
**System information**
- TensorFlow version (you are using): 1.11.0
- Are you willing to contribute it (Yes/No): yes

**Will this change the current api? How?**
not substantially

**Who will benefit with this feature?**
Potentially everybody using a MonitoredTrainingSession

**Any Other info.**
None

**Describe the feature and the current behavior/state.**

I am using tf.MonitoredSession, it happens sometimes that you want to evaluate some nodes but you don't want to proceed in the training step, so you do not want to use the Hooks to log stuffs. It seems that the session is wrapped several times and the raw session is stored in a nested way: https://github.com/tensorflow/tensorflow/issues/8425 but it can also be retrieved with method _tf_sess() https://github.com/tensorflow/tensorflow/issues/11971

To recap I think there are two possible ways to run without hooks (to the best of my knowledge):

1. use step_fn https://www.tensorflow.org/api_docs/python/tf/train/MonitoredSession#run_step_fn
2. access the raw session with raw_session = sess._tf_sess()

I think way 2. is the more intuitive for me if I want to be free from the hooks, but is it proper practice are there some problems in doing this? I wonder why is it private and it is not contemplated to have a method to access the raw session like in SingularMonitoredSession.raw_session()? https://www.tensorflow.org/api_docs/python/tf/train/SingularMonitoredSession 

At line 1010 in https://github.com/tensorflow/tensorflow/blob/r1.11/tensorflow/python/training/monitored_session.py I can see that raw_session is defined exactly returning self._tf_sess(), so I wonder why the explicit choice to not have the user the possibility to access the raw session in MonitoredSession, is it possible to incur in some problems using sess._tf_sess()? I wonder if it has something to do with some problems in the distributed setting..

Summarizing: Would it be possible to have a method to access the raw session in MonitoredSession, are there some counter-indications with this kind of procedure?
(accessing the raw session with sess._tf_sess() )

**Comment by harshini-gadige** [comment]
@ispirmustafa   PTAL

**Comment by tensorflowbutler** [comment]
Nagging Assignee @karmel: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by karmel** [comment]
Closing, as @ispirmustafa provided the preferred method of access.
